### ISSUE TITLE
Dilated Pooling

### ISSUE THREAD

**Issue by psicalculus** [BVLC/caffe#4518]
Could you please add 2D/3D dilated pooling (especially max-pooling) to Caffe?

It's basically the same max-pooling operation but you select pixels/voxels in the same manner as dilated convolution.

**Comment by naibaf7** [comment]
@psicalculus 
There is dilated max-pooling in the OpenCL branch: https://github.com/BVLC/caffe/tree/opencl
If you compile with CUDA and cuDNN instead of OpenCL, and use the integrated pooling layer, you have both 2D/3D convolution and pooling, both with and without dilation.

**Comment by shelhamer** [comment]
It makes sense to include this in master, and it's always been strange to me that only conv is dilated since this doesn't preserve the equivalence of the non-dilated model function. For what it's worth it usually still works empirically.

**Comment by ubuntuslave** [comment]
Hey, guys. Do you know if dilated pooling has been implemented in any branch or fork yet?

**Comment by naibaf7** [comment]
@ubuntuslave Yes, OpenCL branch has dilated pooling for both CUDA and Caffe. There is a very fast CUDA implementation available within LibDNN, which you can just enable in the build configuration.
